src/Controller/CRUDController.php

```diff
@@ -1267,7 +1267,14 @@ protected function redirectTo($object)
         if (!$url) {
             foreach (['edit', 'show'] as $route) {
                 if ($this->admin->hasRoute($route) && $this->admin->hasAccess($route, $object)) {
-                    $url = $this->admin->generateObjectUrl($route, $object);
+                    $tab = $request->request->get('_tab');
+                    $params = [];
+
+                    if ($tab) {
+                        $params['_tab'] = $tab;
+                    }
+
+                    $url = $this->admin->generateObjectUrl($route, $object, $params);
```

Comment:
You can simplify even more:
```suggestion
                    $url = $this->admin->generateObjectUrl(
                        $route, 
                        $object, 
                        $request->request->has('_tab') ? ['_tab' => $request->request->get('_tab')] : []);
```